lib/gps_sim/Kconfig

```diff
@@ -103,6 +114,18 @@ config GPS_SIM_TRIGGER_USE_BUTTON
 
 endchoice
 
+config GPS_SIM_GPIO_CONTROLLER
```

Comment:
This is a type of issue I'm not sure how to resolve. If I place this within `if GPS_SIM_TRIGGER_USE_BUTTON`, the compiler will complain of undefined macros if it's not enabled. The alternative why I've used earlier is to add a hidden config like
```
if !GPS_SIM_TRIGGER_USE_BUTTON
config GPS_SIM_GPIO_CONTROLLER
        string
        default " "
endif
```

Is this actually the way to do it? It looks a bit messy, but I know very little of Kconfig conventions

Reply:
By the way, this is a consequence of using `if (IS_ENABLED(CONFIG_...))` instead of `#if defined` in the code. From previous reviews it seems that is the preferred way to do it.

Reply:
> Is this actually the way to do it? It looks a bit messy, but I know very little of Kconfig conventions

No, that's very hacky. I suggest to only make an option appear when it makes sense based on what options are selected and update the code accordingly. It's not mandatory to use `IS_ENABLED` instead of `#ifdef` but it is preferred when possible; in this case it is not.

Reply:
Thanks for clearing that up! It'll be nice to get rid of these workarounds polluting the Kconfig. I'll remove them and add `depends on` to the options instead, and add `#ifdef`s instead of `IS_ENABLED(...)` in `gps_sim.c`.

By the way, this hackery is probably used elsewhere as well as I was advised to do it during a review some time ago. I'll go through my contributions in the near future and clean up wherever I used it.